pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1194,6 +1206,66 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+func (p *PravegaCluster) ValidateSegmentStore() error {
```

Comment:
Please, add comments in this method explaining what it does and why. Also, the name of the method should be more concrete, like `ValidateSegmentStoreMemorySettings`, as `ValidateSegmentStore` is not really descriptive about what it is doing.